crates/executor/src/blockchain_tree/block_indices.rs

```diff
@@ -0,0 +1,258 @@
+//! Implementation of [`BlockIndices`] related to [`super::BlockchainTree`]
+
+use super::chain::{Chain, ChainId, ForkBlock};
+use reth_primitives::{BlockHash, BlockNumber, SealedBlockWithSenders};
+use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet};
+
+/// Internal indices of the block.
+#[derive(Clone)]
+pub struct BlockIndices {
+    /// Last finalized block.
+    pub last_finalized_block: BlockNumber,
+    /// Index needed when discarding the chain, so we can remove connected chains from tree.
+    /// NOTE: It contains just a blocks that are forks as a key and not all blocks.
+    pub fork_to_child: HashMap<BlockHash, HashSet<BlockHash>>,
+    /// Canonical chain. Contains N number (depends on `finalization_depth`) of blocks.
+    /// These blocks are found in fork_to_child but not inside `blocks_to_chain` or
+    /// `number_to_block` as those are chain specific indices.
+    pub canonical_chain: BTreeMap<BlockNumber, BlockHash>,
+    /// Block hashes and side chain they belong
+    pub blocks_to_chain: HashMap<BlockHash, ChainId>,
+    /* Add additional indices if needed as in tx hash index to block */
+    /// Utility index, Block number to block hash.
```

Comment:
Expand a bit here, what is the utility/what blocks are in here?

Reply:
Expanded doc comment. Additional indices are mostly needed for rpc, which is why it is utility.